grails-app/services/au/org/ala/biocollect/merit/ActivityService.groovy

```diff
@@ -464,4 +465,30 @@ class ActivityService {
             return result.content?.subMap('data')  ?: result
         }
     }
+
+    /**
+     * Get linked entities such as outputs, site and documents for a list of activities.
+     */
+    List<Map> addLinkedEntitiesToActivities(List<Map> activities) {
+        if (activities) {
+            List ids = activities.activityId
+            List<Map> linkedActivities = search([activityId: ids])?.resp?.activities
+            if (!linkedActivities) {
+                return activities
+            }
+
+            activities.each {activity ->
+                Map match = linkedActivities.find { activity.activityId == it.activityId }
+                if (match) {
+                    INCLUDE_LINKED_ENTITIES.each { String entity ->
+                        if (match[entity]) {
+                            activity[entity] = match[entity]
+                        }
+                    }
+                }
+            }
+        }
+
+        activities
```

Comment:
this implicitly return the activities, but i think it's a good idea if we explicitly add the return keyword for better clarity

Reply:
Return statement is required in Java. But not required by Groovy. Earlier return statement was to break the flow of code.